File: source/logging/LogType.java
package logging;

import math.Float3;

import java.awt.*;

public enum LogType {
    INFO,
    DEBUG,
    SCRIPT,
    WARNING,
    ERROR;

    LogType() {}

    private static final Float3 infoColor = new Float3(new Color(215, 215, 215));
    private static final Float3 debugColor = new Float3(new Color(64, 194, 159));
    private static final Float3 scriptColor = new Float3(new Color(203, 189, 89));
    private static final Float3 warningColor = new Float3(new Color(210, 126, 66));
    private static final Float3 errorColor = new Float3(new Color(204, 63, 75));

    public Float3 toColor() {
        return switch (this) {
            case INFO    -> infoColor;
            case DEBUG   -> debugColor;
            case SCRIPT  -> scriptColor;
            case WARNING -> warningColor;
            case ERROR   -> errorColor;
        };
    }

    public String toString() {
        return switch (this) {
            case INFO    -> "INFO   ";
            case DEBUG   -> "DEBUG  ";
            case SCRIPT  -> "SCRIPT ";
            case WARNING -> "WARNING";
            case ERROR   -> "ERROR  ";
        };
    }
}

File: source/logging/LogType.kt
package logging

import math.Float3
import java.awt.Color

enum class LogType {
    INFO, DEBUG, SCRIPT, WARNING, ERROR;

    fun toColor(): Float3 {
        return when (this) {
            INFO -> infoColor
            DEBUG -> debugColor
            SCRIPT -> scriptColor
            WARNING -> warningColor
            ERROR -> errorColor
        }
    }

    override fun toString(): String {
        return when (this) {
            INFO -> "INFO   "
            DEBUG -> "DEBUG  "
            SCRIPT -> "SCRIPT "
            WARNING -> "WARNING"
            ERROR -> "ERROR  "
        }
    }

    companion object {
        private val infoColor = Float3(Color(215, 215, 215))
        private val debugColor = Float3(Color(64, 194, 159))
        private val scriptColor = Float3(Color(203, 189, 89))
        private val warningColor = Float3(Color(210, 126, 66))
        private val errorColor = Float3(Color(204, 63, 75))
    }
}